src/lib/mdns/minimal/records/Txt.h

```diff
@@ -0,0 +1,68 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <string.h>
+
+#include <mdns/minimal/records/ResourceRecord.h>
+
+namespace mdns {
+namespace Minimal {
+
+class TxtResourceRecord : public ResourceRecord
+{
+public:
+    static constexpr uint64_t kDefaultTtl = 10;
+
+    TxtResourceRecord(const FullQName & qName, const char ** entries, size_t entryCount) :
+        ResourceRecord(QType::TXT, qName), mEntries(entries), mEntryCount(entryCount)
+    {
+        SetTtl(kDefaultTtl);
+    }
+
+    template <size_t N>
+    TxtResourceRecord(const FullQName & qName, const char * (&entries)[N]) :
+        ResourceRecord(QType::TXT, qName), mEntries(entries), mEntryCount(N)
+    {
+        SetTtl(kDefaultTtl);
+    }
+
+protected:
+    bool WriteData(chip::BufBound & out) const override
+    {
+        for (size_t i = 0; i < mEntryCount; i++)
+        {
+            size_t len = strlen(mEntries[i]);
```

Comment:
Suggest `strnlen(mEntries[i], (kMaxTxtLength + 1))` with `constexpr size_t kMaxTxtLength = 255`;

The `if (len > 0xff)` could use the constant as well, the line below.

Reply:
done on the constant.

Strlen though seems good enough for readability (the speed gain to restrict on max+1 may not outweigh the readability burden). This is also always guaranteed happy path in production: we will not willingly add invalid entries or if we do, we will fix it fast because otherwise things will be broken and mDNS will not work.